aten/src/ATen/native/nested/NestedTensorMatmul.cpp

```diff
@@ -306,7 +306,7 @@ Tensor matmul_nested(const Tensor& self, const Tensor& mat2) {
       self_dim == 4 && self.is_contiguous() &&
       mat2_dim == 4 && mat2.is_contiguous() &&
       !(GradMode::is_enabled() && (self.requires_grad() || mat2.requires_grad()))) {
-    auto n_heads = self_sizes.select(0, 1).select(0, 0).item<int64_t>();
+    auto n_heads = self_sizes.select(0, 0).select(0, 0).item<int64_t>();
```

Comment:
Is this fixing a bug? If so let's split it out into a separate PR and update the corresponding matmul tests to cover it.

Reply:
Also can't we use opt_dim to do this?